### PR TITLE
Feature/tec 562 add scope request timeout

### PR DESCRIPTION
**Pull request by JustinGuedes**

### CODE REVIEW

File: CivicConnect/Classes/Utilities/RepeatTimer.swift [hunk 1]
```diff
@@ -42,6 +57,23 @@ class RepeatTimer {
         timer = nil
     }
     
+    func executeTimeOutExecutionIfTimedOut() -> Bool {
+        guard var timeOut = timeOut else {
+            return false
+        }
+        
+        print("EXECUTE")
```
Comment (TYRONEMICHAEL) [inline CivicConnect/Classes/Utilities/RepeatTimer.swift]: Is this meant to be in here?

File: Tests/RepeatTimerTests.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    
+    func testShouldInvalidateTimerBeforeExecutingThreeTimes() {
+        var counter = 0
+        let repeatTimer = CivicConnect.RepeatTimer(timeInterval: 0.001) { timer in
+            counter += 1
+            if counter == 2 { timer.invalidate() }
+        }
+        
+        repeatTimer.fire()
+        usleep(5000) // 5ms
```
Comment (leolangen) [inline Tests/RepeatTimerTests.swift]: We know that tests with sleep can be problematic. Can you think of a way of testing this without sleeping?

File: CivicConnect/Classes/Utilities/RepeatTimer.swift (hunk 1 above)
Comment (JustinGuedes) [reply CivicConnect/Classes/Utilities/RepeatTimer.swift]: Nope, it's not. Thanks for picking that up.
```suggestion

```

File: Tests/RepeatTimerTests.swift (hunk 2 above)
Comment (JustinGuedes) [reply Tests/RepeatTimerTests.swift]: Yeah so in this case, we are testing a timer, so without having to do some sort of waiting, how do we know the timer is actually working? I am open to any other suggestions?

**Comment by civictechuser** [comment]
## Current coverage for CivicConnect.framework is `79.18%`
Files changed | - | - 
--- | --- | ---
ConnectError.swift | `55.56%` | :warning:
RepeatTimer.swift | `92.31%` | :white_check_mark:
DefaultConnectSession.swift | `93.64%` | :white_check_mark:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Tests/RepeatTimerTests.swift (hunk 2 above)
Comment (leolangen) [reply Tests/RepeatTimerTests.swift]: Dankie Justice!

**Pull request merged**